stan/math/rev/meta/rev_matrix_type.hpp

```diff
@@ -0,0 +1,38 @@
+#ifndef STAN_MATH_REV_META_REV_MATRIX_TYPE_HPP
+#define STAN_MATH_REV_META_REV_MATRIX_TYPE_HPP
+
+#include <stan/math/prim/meta/disjunction.hpp>
+#include <stan/math/prim/meta/conjunction.hpp>
+#include <stan/math/prim/meta/require_generics.hpp>
+#include <stan/math/prim/meta/return_type.hpp>
+#include <stan/math/prim/fun/Eigen.hpp>
+#include <stan/math/rev/core/var.hpp>
+
+namespace stan {
+
+/**
+ * Determines a return type for a function that addepts given inputs and wants
+ * to return a matrix(or vector or row vector) with given compile time number of
+ * rows and columns.
+ *
+ * Returns `var_value` if any of the inputs is `var_value`, otherwise a matrix
+ * of `var` or `double`.
```

Comment:
```suggestion
 * Determines a return type for a function that accepts given inputs and wants
 * to return a matrix (or vector or row vector) with given compile time number of
 * rows and columns.
 *
 * Returns `var_value` if any of the inputs are `var_value`, otherwise returns an Eigen matrix
 * of `var` or `double`.
```

Reply:
I am typing too fast ... Fixed.